src/shared/components/TokenAmountInput.tsx

```diff
@@ -23,13 +23,15 @@ function handleValidate(
 ): { value: bigint | undefined } | { error: string } {
   const parsed = userAmountToBigInt(value)
 
-  if (parsed !== undefined) {
-    if (parsed < 0n) {
-      return { error: AmountErrors.INVALID_VALUE }
-    }
-    if (parsed > balance) {
-      return { error: AmountErrors.NOT_ENOUGH_FUNDS }
-    }
+  if (
+    (parsed === undefined && value) ||
+    (parsed !== undefined && parsed < 0n)
+  ) {
+    return { error: AmountErrors.INVALID_VALUE }
+  }
+
+  if (parsed !== undefined && parsed > balance) {
```

Comment:
Let's write it down more clearly, group by conditions not by results(more human-readable):

```
if(!parsed){
  if(value} ...
}
if(parsed){
  if(parsed < 0n) ...
  if(parsed > balance) ...
}
```